packages/opentelemetry-web/src/utils.ts

```diff
@@ -38,6 +38,16 @@ const getUrlNormalizingAnchor = () => {
   return a;
 };
 
+export function makeSafe<TThis, TArgs extends any[]>(func: (this: TThis, ...args: TArgs) => void, logger: api.DiagLogger) {
+  return function (this: TThis, ...args: TArgs): void {
+    try {
+      return func.call(this, ...args)
+    } catch (error) {
+      logger.error('Instrumentation error', error)
```

Comment:
It would be worth falling back to api.diag when no logger is provided (ie. someone passes null)